### ISSUE TITLE
Build fails on openssl v0.7.14

### ISSUE THREAD

**Issue by devtea** [khyperia/weechat-discord#18]
I'm finding that rebuilding weechat-discord fails at building openssl.

Currently building 8800f81 

```
$ cargo build                                                                                     
 Downloading libc v0.2.22
 Downloading num_cpus v1.4.0
   Compiling base64-rs v0.1.1
   Compiling itoa v0.1.1
   Compiling httparse v1.2.2
   Compiling matches v0.1.4
   Compiling num-traits v0.1.37
   Compiling pkg-config v0.3.9
   Compiling bitflags v0.7.0
   Compiling semver v0.1.20
   Compiling rustc_version v0.1.7
   Compiling log v0.3.7
   Compiling hpack v0.2.0
   Compiling solicit v0.4.4
   Compiling openssl-sys v0.7.17
   Compiling dtoa v0.2.2
   Compiling traitobject v0.0.1
   Compiling siphasher v0.2.2                                                                                                            Compiling cfg-if v0.1.0
   Compiling language-tags v0.2.2
   Compiling unicode-normalization v0.1.4
   Compiling unicase v1.4.0
   Compiling serde v0.8.23
   Compiling lazy_static v0.2.8
   Compiling libc v0.2.22
   Compiling time v0.1.37
   Compiling num_cpus v1.4.0
   Compiling rand v0.3.15
   Compiling tempdir v0.3.5
   Compiling net2 v0.2.27
   Compiling byteorder v0.5.3
   Compiling rustc-serialize v0.3.24
   Compiling typeable v0.1.2
   Compiling unicode-bidi v0.2.5
   Compiling mime v0.2.3
   Compiling phf_shared v0.7.21
   Compiling phf_generator v0.7.21
   Compiling phf_codegen v0.7.21
   Compiling phf v0.7.21
   Compiling mime_guess v1.8.1
   Compiling gcc v0.3.45
   Compiling openssl-sys-extras v0.7.14
error: failed to run custom build command for `openssl-sys-extras v0.7.14`
process didn't exit successfully: `/home/user/build/weechat-discord/target/debug/build/openssl-sys-extras-f9e8037051c16bfa/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-unknown-linux-gnu = None                                                                                                    CC_x86_64_unknown_linux_gnu = None
HOST_CC = None
CC = None
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CFLAGS_x86_64-unknown-linux-gnu = None
CFLAGS_x86_64_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
PROFILE = Some("debug")
running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-m64" "-o" "/home/eytosh/build/weechat-discord/target/debug/build/openssl-sys-extras-8e2dc543f8892677/out/src/openssl_shim.o" "-c" "src/openssl_shim.c"
cargo:warning=src/openssl_shim.c: In function ‘DH_new_from_params’:
cargo:warning=src/openssl_shim.c:132:7: error: dereferencing pointer to incomplete type ‘DH {aka struct dh_st}’
cargo:warning=     dh->p = p;
cargo:warning=       ^~
cargo:warning=src/openssl_shim.c: In function ‘X509_get_extensions_shim’:
cargo:warning=src/openssl_shim.c:143:13: error: dereferencing pointer to incomplete type ‘X509 {aka struct x509_st}’
cargo:warning=     return x->cert_info ? x->cert_info->extensions : NULL;
cargo:warning=             ^~
ExitStatus(ExitStatus(256))


command did not execute successfully, got: exit code: 1
                                                                                                                                      

--- stderr
thread 'main' panicked at 'explicit panic', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.45/src/lib.rs:1152
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by devtea** [comment]
Seeing the same issue when trying to build 9863c11 or 2922026

**Comment by devtea** [comment]
Looking at weechat, the existing plugin (that I'm trying to rebuild) is reporting the following error on loading:

> =!= Error: unable to load plugin "/home/user/.weechat/plugins/libweecord.so": libssl.so.1.0.0: cannot open shared object file: No such file or directory

Currently installed OpenSSL version is openssl 1.1.0.e-1
Installing openssl 1.0.2 changes the error message in weechat to

> Fatal error (caught) - called `Result::unwrap()` on an `Err` value: OpenSslErrors([UnknownError { library: "SSL routines", function: "SSL_CTX_new", reason: "library has no ciphers" }])

**Comment by khyperia** [comment]
This should be resolved by c0041f87a359ce316758fe4b75fba96cc8bf51f7. Thanks for the info and issue report to track this :)
